core/src/main/scala/org/ensime/indexer/ClassfileIndexer.scala

```diff
@@ -187,6 +187,16 @@ trait ClassfileIndexer {
         case member => FieldName(ClassName.fromInternal(owner), member)
       }
 
+    private def extractClassNames(className: String): Seq[String] = {
```

Comment:
I'm not entirely sure what this method does, but it seems like it might be a memory hog.